### PR TITLE
[FX-4607] Migrate ApplicationUpdateNotification to TailwindCSS

### PR DESCRIPTION
**Pull request by AdrianContiu**
[FX-4607]

### Description

This PR changes the ApplicationUpdateNotification component to use the Snackbar from the MUI Base UI and applies styles using TailwindCSS.

### How to test

- [Temploy](https://picasso.toptal.net/fx-4607-migrate-applicationupdatenotification)
- Check the [production](https://picasso.toptal.net/?path=/story/widgets-applicationupdatenotification--applicationupdatenotification) and compare it with the temploy

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Picasso - ApplicationUpdateNotification - Google Chrome 2024-05-02 at | Picasso - ApplicationUpdateNotification - Google Chrome 2024-05-02 at  (1) |

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- ~Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)~
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- ~Annotate all `props` in component with documentation~
- ~Create `examples` for component~
- ~Ensure that deployed demo has expected results and good examples~
- ~Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).~
- [x] Self reviewed
- ~Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)~

**Breaking change**

- ~codemod is created and showcased in the changeset~
- ~test alpha package of Picasso in StaffPortal~

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4607]: https://toptal-core.atlassian.net/browse/FX-4607?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by AdrianContiu** [comment]
@toptal-anvil ping reviewers

File: packages/base/ApplicationUpdateNotification/src/ApplicationUpdateNotification/ApplicationUpdateNotification.tsx [hunk 1]
```diff
@@ -62,22 +54,24 @@ export const ApplicationUpdateNotification = forwardRef<HTMLElement, Props>(
             variant='heading'
             size='medium'
             color='inherit'
-            className={classes.positionRelative}
+            className={'relative'}
```
Comment (mkrl) [inline packages/base/ApplicationUpdateNotification/src/ApplicationUpdateNotification/ApplicationUpdateNotification.tsx:57]:
```suggestion
            className='relative'
```
For all the casesd like this (static strings in prop values), you can just leave them without the `{` and `}` (according to our style guides). I think the ESLint / prettier is supposed to auto-fix that, maybe it's not running or it became broken in our setup?

**Review by mkrl — APPROVED** [review]
Nice job! :+1: Tested on temploy, looks 1:1. Left one note that's most likely related to the dev setup

File: packages/base/ApplicationUpdateNotification/src/ApplicationUpdateNotification/ApplicationUpdateNotification.tsx [hunk 2]
```diff
@@ -62,22 +54,24 @@ export const ApplicationUpdateNotification = forwardRef<HTMLElement, Props>(
...
           >
             {title}
           </Typography>
           {dismissable && (
             <ButtonCircular
               data-testid='btn-dismiss'
-              className={classes.closeAction}
+              className={
+                'text-lg text-white inline-flex items-center font-semibold whitespace-nowrap'
+              }
```
Comment (dmaklygin) [inline packages/base/ApplicationUpdateNotification/src/ApplicationUpdateNotification/ApplicationUpdateNotification.tsx:66]: braces are odd here

File: packages/base/ApplicationUpdateNotification/src/ApplicationUpdateNotification/__snapshots__/test.tsx.snap [hunk 3]
```diff
@@ -6,43 +6,39 @@ exports[`ApplicationUpdateNotification renders 1`] = `
     class="Picasso-root"
   >
     <div
-      class="MuiPaper-root MuiSnackbarContent-root ApplicationUpdateNotification-root ApplicationUpdateNotification-positionRelative MuiPaper-elevation6"
-      role="alert"
+      class="base-Snackbar relative flex flex-wrap text-lg/[1.43] items-center rounded-md bg-blue shadow-3 w-[27.5em] text-white p-[1.5em] overflow-hidden"
+      role="presentation"
```
Comment (dmaklygin) [inline packages/base/ApplicationUpdateNotification/src/ApplicationUpdateNotification/__snapshots__/test.tsx.snap:10]: role has been changed. is it intentional?

**Review by dmaklygin — COMMENTED** [review]
Left some commentes

File: packages/base/ApplicationUpdateNotification/src/ApplicationUpdateNotification/ApplicationUpdateNotification.tsx (hunk 1 above)
Comment (AdrianContiu) [reply packages/base/ApplicationUpdateNotification/src/ApplicationUpdateNotification/ApplicationUpdateNotification.tsx:57]: I checked with Tomas and couldn't find any settings on ESLint regarding this topic. However, I should start remembering to stop adding the brackets.

File: packages/base/ApplicationUpdateNotification/src/ApplicationUpdateNotification/__snapshots__/test.tsx.snap (hunk 3 above)
Comment (AdrianContiu) [reply packages/base/ApplicationUpdateNotification/src/ApplicationUpdateNotification/__snapshots__/test.tsx.snap:10]: No, this was not intentional. I'm not sure why it happend. What do you recommend me to do?

Comment (TomasSlama) [reply packages/base/ApplicationUpdateNotification/src/ApplicationUpdateNotification/__snapshots__/test.tsx.snap:10]: as discussed yesterday, we need to understand why we have this change and if it is a mistake or by purpose. Ping me if you will need help with that.

File: .changeset/five-beds-wonder.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+'@toptal/picasso-application-update-notification': major
+---
+
+### ApplicationUpdateNotification
+
+- migrate from MUI to TailwindCSS and MUI Base UI
+- row attribute is changed from 'alert' to presentation due an issue with the screen readers in Firefox https://github.com/mui/material-ui/issues/29080
```
Comment (TomasSlama) [inline .changeset/five-beds-wonder.md:8]:
```suggestion
- change the `role` attribute from `alert` to `presentation` due to an issue with the screen readers in Firefox https://github.com/mui/material-ui/issues/29080
```

File: packages/base/ApplicationUpdateNotification/src/ApplicationUpdateNotification/__snapshots__/test.tsx.snap (hunk 3 above)
Comment (AdrianContiu) [reply packages/base/ApplicationUpdateNotification/src/ApplicationUpdateNotification/__snapshots__/test.tsx.snap:10]: [Here](https://github.com/toptal/picasso/pull/4278#discussion_r1593913218) is the explanation of why the role has been changed.

Comment (AdrianContiu) [reply packages/base/ApplicationUpdateNotification/src/ApplicationUpdateNotification/__snapshots__/test.tsx.snap:10]: With Tomas's help, I dug up and the change to the role attribute was done due to a problem with screen reader in Firefox. More details can be found [here](https://github.com/mui/material-ui/issues/29080). I also updated the changeset file to include this.

**Review by TomasSlama — APPROVED** [review]

**Pull request merged**